### PR TITLE
Reworked the remote authentication

### PR DESCRIPTION
**Pull request by annando**
Fixes https://github.com/friendica/friendica/issues/7657

This started as some small error correction, but turned out to be "a little bit" bigger.

There is still work left to do, like moving the "redir" functionality into the class structure. I also want to have a look at improving the "magic auth" code a little bit. And also I guess that there are other things out there that still needs to be fixed.

### CODE REVIEW

**Comment by nupplaphil** [comment]
Just for information: On my roadmap for 2019.12 or 2020.03 is to get rid of the `local_user()` and `remote_user()` function (and their combos..) and to create a `CurrentUser` object, created by the `Session` class, which holds its `isRemote()`, `isLocal()`, `isAuthenticated()` states. This `CurrentUser` would than be available with `self::getClass(CurrentUser::class)` or with Dependency injection per constructor. Or it could be directly used for tests without a session behind it ;)

But currently I'm working on other areas :) so just for in for for this PR

**Comment by annando** [comment]
@nupplaphil concerning ```isAuthenticated``` it seems that we had the same idea. But with this "injection" thing, I guess the handling will be "slightly" different then ...

BTW: I still don't have a clue how to handle this stuff. I read the comment ```use BaseURL->getScheme() instead```in the source code and was absolutely clueless what "BaseURL" would look like in a real example.

**Comment by nupplaphil** [comment]
Basically we're currently using both static reference and Dependency Injection. Therefore I created a global registry for classes - which are normally dependency injected - inside the `BaseObject` class.

So for example if you want to get the Scheme of the Base URL, you need to exend from the class `BaseObject` and get the class with:
```php
/** @var BaseUrl */
$baseUrl = self::getClass(BaseUrl::class);
$baseUrl->getScheme();
```

With Dependency Injection, the `$baseUrl` would be set over the constructor with
```php
class AClass
{
    /** @var BaseUrl */
    private $baseUrl;

   public function __construct(BaseUrl $baseUrl)
   {
      $this->baseUrl = $baseUrl;
   }

   public function aMethod()
   {
      $this->baseUrl->getScheme();
   }
}
```
which will get injected automatically by Dice (because of the `__construct` constructor information, Dice checks that `BaseUrl` is needed and creates it for `AClass`).

**Comment by nupplaphil** [comment]
Originally, I wanted to first create a `CurrentProfile` class for all the infos about the shown profile and than a `CurrentUser` class for all the infos about the current user. So we focus the infos a little bit more.

**Comment by annando** [comment]
It's still too high for me. So for example I'm somewhere in some addon. What do I have to type in to call that function?

I totally understand the motivation behind this - but I simply don't get it how to work with this.

But something different: I wanted to change the mod/redir.php into some class. But since PR https://github.com/friendica/friendica/pull/7670 is currently pending, I would do this as a later point in time in a separate PR.

File: mod/display.php [hunk 1]
```diff
@@ -84,10 +87,6 @@ function display_init(App $a)
 		displayShowFeed($item['id'], $a->argc > 3 && $a->argv[3] == 'conversation.atom');
 	}
 
-	if ($a->argc >= 3 && $nick == 'feed-item') {
-		displayShowFeed($item['id'], $a->argc > 3 && $a->argv[3] == 'conversation.atom');
```
Comment (MrPetovan) [inline mod/display.php]: Why did you remove this?

File: mod/videos.php [hunk 2]
```diff
@@ -114,7 +111,7 @@ function videos_content(App $a)
 	// videos/name/video/xxxxx/edit
 
 
-	if ((Config::get('system', 'block_public')) && (!local_user()) && (!remote_user())) {
+	if (Config::get('system', 'block_public') && !Session::isAuthenticated()) {
 		notice(L10n::t('Public access denied.') . EOL);
```
Comment (MrPetovan) [inline mod/videos.php]: This should throw a `ForbiddenException` like this:
```php
throw new HTTPException\ForbiddenException(L10n::t('Public access denied.'));
```

File: mod/display.php (hunk 1 above)
Comment (annando) [reply mod/display.php]: Look at the lines above - I guess some merge accidentally duplicated these lines.

File: mod/videos.php (hunk 2 above)
Comment (annando) [reply mod/videos.php]: When we decide to do it this way, we have to do it at all places.

Comment (MrPetovan) [reply mod/videos.php]: Fair enough.

Comment (MrPetovan) [reply mod/videos.php]: Fair enough.

Comment (annando) [reply mod/videos.php]: If you had said: "Do it so", then I would had changed it at every place :-)

Comment (MrPetovan) [reply mod/videos.php]: But it wouldn't have been in the scope of this already large PR.
